be/src/runtime/routine_load/data_consumer.cpp

```diff
@@ -253,13 +253,39 @@ Status KafkaDataConsumer::get_partition_offset(std::vector<int32_t>* partition_i
     for (auto p_id : *partition_ids) {
         int64_t beginning_offset;
         int64_t latest_offset;
-        RdKafka::ErrorCode err = _k_consumer->query_watermark_offsets(_topic, p_id, &beginning_offset, &latest_offset,
-                                                                      config::routine_load_kafka_timeout_second * 1000);
+
+        //When all kafka brokers are down,  query_watermark_offsets would block until timeout, which make the bthread unable to be preemptive.
+        // Here, we make 2 attempt to query_watermark_offsets.
+        // When all brokers are down, the 1st query_watermark_offsets with 1s timeout would fail quickly and return RdKafka::ERR__ALL_BROKERS_DOWN.
+        // When the 1st query_watermark_offsets fails but not due to all brokers are down, the 2nd query_watermark_offsets would try with timeout config::routine_load_kafka_timeout_second.
+
+        // fast path
+        RdKafka::ErrorCode err =
+                _k_consumer->query_watermark_offsets(_topic, p_id, &beginning_offset, &latest_offset, 1000 /*1000ms*/);
```

Comment:
The number of Bthreads of Brpc is limited, = the number of cores. If it is stuck for one second, will the problem also be serious

Reply:
Is it possible to change to asynchronous？